web-local/src/lib/application-state-stores/metrics-internal-store.ts

```diff
@@ -152,6 +170,58 @@ export class MetricsInternalRepresentation {
     this.regenerateInternalYAML();
   }
 
+  updateErrors(errors: Array<V1ReconcileError>) {
+    const measureErroredIndices = new Set<number>();
+    const dimensionErroredIndices = new Set<number>();
+    // set errors for measures and dimensions
+    for (const error of errors) {
+      const index = Number(error.propertyPath[1]);
+      switch (error.propertyPath[0]) {
+        case "Measures":
+          measureErroredIndices.add(index);
+          this.internalRepresentationDocument.setIn(
+            ["measures", index, "__ERROR__"],
+            error.message
+          );
+          break;
+        case "Dimensions":
+          dimensionErroredIndices.add(index);
+          this.internalRepresentationDocument.setIn(
+            ["dimensions", index, "__ERROR__"],
+            error.message
+          );
+          break;
+      }
+    }
+
+    // remove previous errors if set
```

Comment:
Wouldn't it be more clear and concise to _first_ remove all existing errors, _then_ set errors?

Reply:
You are right. An instance of this class is created and the yaml documented regenerated every time. So this is not needed at all.